bokeh/util/serialization.py

```diff
@@ -189,17 +195,30 @@ def transform_numerical_array(obj):
         transformed[np.isneginf(obj)] = '-Infinity'
         return transformed.tolist()
 
-def traverse_data(datum):
-    """recursively dig until a flat array is found
-    convert it to a numpy array and send off to transform_array()
-    to handle nan, inf, -inf
+def traverse_data(datum, use_numpy=True):
+    """recursively dig until a flat list is found
+    if numpy is available convert the flat list to a numpy array
+    and send off to transform_array() to handle nan, inf, -inf
+    otherwise iterate through items in array converting non-json items
     """
-    if not any(isinstance(el, (list, tuple)) for el in datum):
+    global is_numpy
```

Comment:
mmm... I don't like globals...

what about 

```python
def traverse_data(datum, is_numpy,  use_numpy=True):

...

if is_numpy and use_numpy and not any(isinstance(el, (list, tuple)) for el in datum):
```

Also make clear in the docstring that you are using `use_numpy` for testing purposes... in fact, I think there is a way to mock the import... but I don't have play enough with that to make it quickly...